.github/workflows/ci.yaml

```diff
@@ -15,20 +19,136 @@ on:
       - cryostat-v[0-9]+.[0-9]+
 
 jobs:
-  build:
+  build-core:
+    runs-on: ubuntu-latest
+    outputs:
+      test_version: ${{ steps.get_core_version.outputs.test_version }}
+    steps:
+    - uses: actions/checkout@v4
+    - id: get_core_version
+      run: |
+        echo "test_version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_OUTPUT
+        echo "name=$(mvn help:evaluate -Dexpression=project.artifactId -q -DforceStdout)" >> $GITHUB_OUTPUT
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: 'maven'
+    - run: mvn -B -U clean install
+    - uses: actions/upload-artifact@v3
+      with:
+        name: cryostat-core
+        path: ${{ github.workspace }}/target/${{ steps.get_core_version.outputs.name }}-${{ steps.get_core_version.outputs.test_version }}.jar
 
+  build-cryostat:
+    needs: [build-core]
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    uses: ./.github/workflows/cryostat-build.yml
+    with:
+      build-arch: ${{ matrix.arch }}
+      test-version: ${{ needs.build-core.outputs.test_version }}
+  
+  push-to-ghcr:
     runs-on: ubuntu-latest
+    needs: [build-cryostat]
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    outputs: 
+      amd64_image: ${{ steps.amd64_image.outputs.image }}
+      arm64_image: ${{ steps.arm64_image.outputs.image }}
+    permissions:
+      packages: write
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: cryostat-${{ matrix.arch }}
+    - name: Load cryostat image
+      run: podman load -i cryostat-${{ matrix.arch }}.tar
+    - name: Tag cryostat image
+      run: podman tag cryostat ghcr.io/${{ github.repository_owner }}/cryostat-core:pr-${{ github.event.number }}-${{ github.event.pull_request.head.sha }}-linux-${{ matrix.arch }}
+    - name: Push PR test image to ghcr.io
+      id: push-to-ghcr
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: cryostat-core
+        tags: pr-${{ github.event.number }}-${{ github.event.pull_request.head.sha }}-linux-${{ matrix.arch }}
+        registry: ghcr.io/${{ github.repository_owner }}
+        username: ${{ github.event.pull_request.user.login }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Store images as output amd64
+      if: ${{ matrix.arch == 'amd64' }}
+      id: amd64_image
+      run: z "image=${{ steps.push-to-ghcr.outputs.registry-path }}" >> "$GITHUB_OUTPUT"
+    - name: Store images as output
+      if: ${{ matrix.arch == 'arm64' }}
+      id: arm64_image
+      run: echo "image=${{ steps.push-to-ghcr.outputs.registry-path }}" >> "$GITHUB_OUTPUT"
 
+  cryostat-test:
+    runs-on: ubuntu-latest
+    needs: [push-to-ghcr]
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-java@v2
+    - name: Install xpath
```

Comment:
Can we use `mvn`? I think the integration script should check for `xpath` and fall back to `mvn`?